src/modm/driver/inertial/bno055.hpp

```diff
@@ -0,0 +1,660 @@
+		Interrupt_t >;
+
+public:
+	enum class
+	TemperatureSource : uint8_t
+	{
+		Accelerometer = 0,
+		Gyroscope = 1,
+	};
+
+	enum class
+	OperationMode : uint8_t
+	{
+		ConfigMode = 0b0000,
+		Acc = 0b0001,
+		Mag = 0b0010,
+		Gyro = 0b0011,
+		AccMag = 0b0100,
+		AccGyro = 0b0101,
+		MagGyro = 0b0110,
+		AMG = 0b0111,
+		IMU = 0b1000,
+		Compass = 0b1001,
+		M4G = 0b1010,
+		NDOF_FMC_Off = 0b1011,
+		NDOF = 0b1100,
+	};
+
+	enum class
+	PowerMode : uint8_t
+	{
+		Normal = 0,
+		LowPower = 1,
+		Suspend = 2,
+	};
+
+	enum class
+	StatusCode : uint8_t
+	{
+		SystemIdle = 0,
+		SystemError = 1,
+		InitializingPeripherals = 2,
+		SystemInitialization = 3,
+		ExecutingSelftest = 4,
+		SensorFusionAlgorithmRunning = 5,
+		SystemRunningWithoutFusionAlgorithm = 6,
+	};
+
+	enum class
+	StatusError : uint8_t
+	{
+		NoError = 0,
+		PeripheralInitializationError = 1,
+		SystemInitializationError = 2,
+		SelfTestResultFailed = 3,
+		RegisterMapValueOutOfRange = 4,
+		RegisterMapAddressOutOfRange = 5,
+		RegisterMapWriteError = 6,
+		BNOLowPowerModeNotAvailableForSelectedOperationMode = 7,
+		AccelerometerPowerModeNotAvailable = 8,
+		FusionAlgorithmConfigurationError = 9,
+		SensorConfigurationError = 0xA,
+	};
+
+public:
+	struct modm_packed
+	Data
+	{
+		template < class I2cMaster >
+		friend class Bno055;
+
+		inline Vector3f acceleration()        const { return Vector3f(r.acceleration).scaled(1.f/16); }
+		inline Vector3f magnetometer()        const { return Vector3f(r.magnetometer).scaled(1.f/16); }
+		inline Vector3f gyroscope()           const { return Vector3f(r.gyroscope).scaled(1.f/16); }
+		inline float heading()                const { return r.heading / 16.f; }
+		inline float roll()                   const { return r.roll / 16.f; }
+		inline float pitch()                  const { return r.pitch / 16.f; }
+		inline Quaternion<float> quaternion() const { return Quaternion<float>(r.quaternion).scale(1l << 14); }
+		inline Vector3f linear_acceleration() const { return Vector3f(r.linear_acceleration).scaled(1.f/100); }
+		inline Vector3f gravity()             const { return Vector3f(r.gravity).scaled(1.f/100); }
+		inline int8_t temperature()           const { return r.temperature; }
+
+	private:
+		struct Raw {
+			int16_t acceleration[3];
+			int16_t magnetometer[3];
+			int16_t gyroscope[3];
+			int16_t heading;
+			int16_t roll;
+			int16_t pitch;
+			int16_t quaternion[4];
+			int16_t linear_acceleration[3];
+			int16_t gravity[3];
+			int8_t temperature;
+		} r;
+		static constexpr uint8_t size = sizeof(Raw);
+	};
+}; // struct bno055
+
+/**
+ * @ingroup modm_driver_bno055
+ * @author	Niklas Hauser
+ */
+template < class I2cMaster >
+class Bno055 : public bno055, public modm::I2cDevice<I2cMaster, 2>
+{
+public:
+	/// Constructor, requires a bno055::Data object.
+	/// For I2c this also sets the address to 0b110101 (alternative: 0x1C).
+	inline Bno055(Data &data, uint8_t address=addr()):
+		I2cDevice<I2cMaster,2>(address), data(data) {}
```

Comment:
`data(data)` is very error prune, as the compiler will still be happy if you rename the argument but don't change this line. Suggest either an `m` prefix for the member or `_` suffix for the parameter, that seem to be the most popular options. Or rename either of the two to something different.

Reply:
Yeah, I’d like to use the Qt coding conventions in general: `m_` prefix and just `obj.data()` as getter. But, yeah, conventions…